### PR TITLE
Add Analytics for PayPal Native Checkout

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- add analytics events for PayPal Native Checkout
- add tests for analytics

### Checklist

- ~[ ] Added a changelog entry~

- @jaxdesmarais

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift [hunk 1]
```diff
@@ -31,7 +31,9 @@ import PayPalCheckout
         with nativeRequest: BTPayPalRequest,
         completion: @escaping (BTPayPalNativeCheckoutAccountNonce?, Error?) -> Void
     ) {
+        apiClient.sendAnalyticsEvent("ios.paypal-native-checkout.tokenize.started")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift:34]: Can we align with the existing BTNXO analytic event naming on Android? [Example here](https://github.com/braintree/braintree_android/blob/6e003485313db9eafe32fb55966ddeed444da46f/PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java#L113). 

So `paypal-native` instead of `paypal-native-checkout`? This will help us when searching for analytics across platforms.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift [hunk 2]
```diff
@@ -40,6 +42,7 @@ import PayPalCheckout
         orderCreationClient.createOrder(with: request) { [weak self] result in
             switch result {
             case .success(let order):
+                self?.apiClient.sendAnalyticsEvent("ios.paypal-native-checkout.create-order.started")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift:45]: Also, it's ideal that we have the same analytic logging b/w platforms per payment method feature/flow. Not just in terms of naming, but also in terms of timing of analytic sendoff in the flow.

This PR adds more events than currently exist for Android NXO flow. This is a good opportunity to make sure we are tracking the same events across both platforms.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift [hunk 3]
```diff
@@ -28,16 +28,19 @@ class BTPayPalNativeOrderCreationClient {
     ) {
         apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
             guard let config = configuration, error == nil else {
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native-checkout.create-order.fetch-configuration.failure")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift:31]: I don't think we need any config fetch analytics. None of our other payment method features have them, since the actual config fetch call already has analytics baked into it.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift (hunk 1 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift:34]: Good call - will update this

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift (hunk 2 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift:45]: Yeah I think we need a follow up ticket on the Android side to add additional events. Right now, we don't track success or failure at all for NXO on Android, just which flow was started. 🙈 I'll create an Android follow up ticket for an audit on that platform for NXO analytics events.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift [hunk 4]
```diff
@@ -33,15 +33,18 @@ class BTPayPalNativeTokenizationClient {
         ) { body, _, error in
             guard let json = body, error == nil else {
                 let underlyingError = error ?? BTPayPalNativeError.invalidJSONResponse
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.failure")
                 completion(.failure(.tokenizationFailed(underlyingError)))
                 return
             }
 
             guard let accountNonce = BTPayPalNativeCheckoutAccountNonce(json: json) else {
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.parsing-result.failure")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift:42]: I don't think we need to get this nuanced with the failure analytics. I think `"ios.paypal-native.tokenize.failed"` is sufficient for all cases. Especially since we're already returning specific errors to the merchant for these error use cases - we will be able to learn from their integration which error they hit.

I think at the analytics level, we just want to know if it was success or not. Historically, we have over-complicated our analytics logging resulting in too many events with different names that we lose track of.

Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift:42]: We also created this [naming guide](https://github.braintreeps.com/team-sdk/sdk-knowledge/blob/main/analytics/events/event-names.md) to try to remedy some of that, and create some uniformity amongst platforms.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift [hunk 5]
```diff
@@ -33,11 +33,13 @@ class BTPayPalNativeOrderCreationClient {
             }
 
             guard config.json["paypalEnabled"].isTrue else {
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.create-order.paypal-not-enabled.failure")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift:36]: See [naming guide](https://github.braintreeps.com/team-sdk/sdk-knowledge/blob/main/analytics/events/event-names.md), prefer past tense such as `failed`.

Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift:36]: Also similar comment here regarding just needing `ios.paypal-native.create-order. succeeded` and `...failed` cases.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift (hunk 4 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift:42]: > I don't think we need to get this nuanced with the failure analytics.

This came up during the merchant call for caching errors and the analytics team we worked with during post mortem recommended us getting more nuanced with the events to better track where we are failing when working with those teams. I don't really have a strong preference one way or the other though. 🤷

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift (hunk 5 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift:36]: Gotcha - can update this, had not seen this before, was just going off of venmo which uses success/failure

Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift]: Sounds good!

Yeh, there has been a lot of inconsistency with naming. We did an audit awhile ago and looked for the most common naming convention used and decided we should try to use that going forward. Re-naming existing analytics isn't usually worth it though because then we have 2 event names in the wild that mean the same thing, and that gets 🤯.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift (hunk 4 above)
Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift]: Ah gotcha! Which post mortem was it for just curious? Which situation caused us to get that feedback?

Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift]: But if that's the feedback we got - makes sense to take it 👍

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift (hunk 2 above)
Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift]: Good spot!

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift (hunk 5 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift]: Ah, that makes sense - I am glad we have a guide for this!

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift (hunk 4 above)
Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift]: @hollabaq86 - could we get your opinion here? I know you've been doing a handful of analytics work on open dev.

TLDR - do we favor more nuanced analytics events such as having several `ios.paypal-native.tokenize.<SPECIFIC_FAILURE_REASON>.failed` events, or instead favor just one `ios.paypal-native.tokenize.failed` event name that covers all tokenization failures?

Comment (hollabaq86) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift]: given that this is one of our larger payment methods, let's go granular and cite the specific failure reason if possible. Thanks for pinging me for my thoughts!

Comment (scannillo) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift]: 👍 Let's do it!

**Comment by jaxdesmarais** [comment]
The UI test `testTokenizeVenmo_whenSignInSuccessfulWithPaymentContext_returnsToApp` is failing in main as well as this branch. Digging into what may have changed to cause this test to succeed yesterday but not today. Since it's also happening on main it is not related to the changes in this PR.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift [hunk 6]
```diff
@@ -62,6 +65,7 @@ class BTPayPalNativeOrderCreationClient {
             ) { json, response, error in
                 guard let hermesResponse = BTPayPalNativeHermesResponse(json: json), error == nil else {
                     let underlyingError = error ?? BTPayPalNativeError.invalidJSONResponse
+                    self.apiClient.sendAnalyticsEvent("ios.paypal-native.create-order.order-creation.failed")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeOrderCreationClient.swift:68]: `.create-order.order-creation.failed` is a little confusing. Maybe `.create-order.hermes-request.failed`?

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift [hunk 7]
```diff
@@ -33,15 +33,18 @@ class BTPayPalNativeTokenizationClient {
         ) { body, _, error in
             guard let json = body, error == nil else {
                 let underlyingError = error ?? BTPayPalNativeError.invalidJSONResponse
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.failed")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift:36]: Similar comment to above. If the other events are granular, this one also might as well be?

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift [hunk 8]
```diff
@@ -33,15 +33,18 @@ class BTPayPalNativeTokenizationClient {
         ) { body, _, error in
             guard let json = body, error == nil else {
                 let underlyingError = error ?? BTPayPalNativeError.invalidJSONResponse
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.request-to-tokenize.failed")
```
Comment (scannillo) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift:36]: Nitpick

```suggestion
                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.tokenize-url-request.failed")
```

**Review by scannillo — APPROVED** [review]
